src/SplitPackages/Program.cs

```diff
@@ -0,0 +1,288 @@
+                }
+            }
+        }
+
+        private static void SetDestinationPath(Arguments arguments, IList<PackageItem> expandedPackages)
+        {
+            foreach (var package in expandedPackages)
+            {
+                var path = Path.Combine(
+                    arguments.DestinationFolder,
+                    package.DestinationFolderName,
+                    package.Name);
+
+                package.DestinationPath = path;
+            }
+        }
+
+        private static void CopyPackages(IList<PackageItem> expandedPackages, bool whatif)
+        {
+            foreach (var package in expandedPackages)
+            {
+                WriteInformational($@"Copying package {package.Name} from
+{package.OriginPath} to
+{package.DestinationPath}");
+                if (!whatif)
+                {
+                    File.Copy(package.OriginPath, package.DestinationPath, overwrite: true);
+                }
+            }
+        }
+
+        private static void WriteInformational(string msg)
+        {
+            Console.ForegroundColor = ConsoleColor.Cyan;
+            Console.WriteLine($"[informational]: {msg}");
+            Console.ResetColor();
+        }
+
+        private static void WriteWarning(string msg)
+        {
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.WriteLine($"[warning]: {msg}");
+            Console.ResetColor();
+        }
+
+        private static void WriteError(string error)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($@"There was an error executing the command:
+{error}");
+            Console.ResetColor();
+            PrintUsage();
+        }
+
+        private static void PrintUsage()
+        {
+            var text = @"
+SplitPackages [help] [source packages] [csv path] [destination folder]
+
+-help Prints out this summary.
+
+-source packages: The path to the folder containing the built packages.
+
+-csv path: The path to the CSV containing the packages and the destination
+           folder for each package.
+
+-destination folder: The path to the folder in which the destination folder for
+           individual package categories will be created.";
+
+            Console.WriteLine(text);
+        }
+
+        private class PackageItem : IEquatable<PackageItem>
+        {
+            public string Name { get; set; }
+            public string OriginPath { get; set; }
+            public string DestinationFolderName { get; set; }
+            public string DestinationPath { get; set; }
+            public bool FoundInCsv { get; set; }
+            public bool FoundInFolder { get; set; }
+
+            public bool Equals(PackageItem other)
+            {
+                return string.Equals(OriginPath, other?.OriginPath);
+            }
+
+            public override int GetHashCode()
+            {
+                return OriginPath?.GetHashCode() * 7 ?? 1;
```

Comment:
As a general rule, avoid being fancy here. Just return the string's hash code and be done.